### ISSUE TITLE
Modify permutations function to take a size argument

### ISSUE THREAD

**Issue by kmsquire** [JuliaLang/julia#12863]
As seen [here](http://stackoverflow.com/questions/32280481/permutations-with-a-limited-length-in-julia):

> In Python, you can use itertools to generate permutations like so:

>     >>> list(itertools.permutations("ABC", 2))
     [('A', 'B'), ('A', 'C'), ('B', 'A'), ('B', 'C'), ('C', 'A'), ('C', 'B')]
     Julia has a similar permutations function, but it only accepts one argument.

It woulds be a reasonable extension of the `permutations` function to add this.

**Comment by IainNZ** [comment]
Why is it reasonable?

**Comment by kmsquire** [comment]
This functionality is something that at least I have used in the past (in Python), and it should be a fairly straightforward extension to the existing functionality. 

Why wouldn't it be reasonable?  I'm open to counter arguments.

**Comment by kmsquire** [comment]
I do also think that the permutations function should probably be moved out of `Base` (assuming it isn't used, of course).

**Comment by IainNZ** [comment]
It just seems like one more thing to add for the sake of it, when it should really be in a package (or you can do what I proposed on SO).

**Comment by IainNZ** [comment]
I don't have strong feelings either way to be honest, but I do lean towards not adding it to Base. Certainly think it should be questioned.

**Comment by kmsquire** [comment]
> (or you can do what I proposed on SO)

Sure.

```julia
julia> collect(chain([permutations(x) for x in subsets(collect("ABC"), 2)]...))
6-element Array{Array{Char,1},1}:
 ['A','B']
 ['B','A']
 ['A','C']
 ['C','A']
 ['B','C']
 ['C','B']
```

I think that would take a little while for most users to figure out if they aren't familiar with the Iterators.jl package.  And I find it rather ugly.  

(But I really would have no problem if the functionality suggested here were added there instead.)

**Comment by kmsquire** [comment]
(But if it's moved to Iterators.jl, I think it would make sense to move the current `permutations` function at the same time.)

**Comment by mschauer** [comment]
Please, permutations certainly belong to base. As a data point: ``permute``  is one of the selected primitives in J, http://www.jsoftware.com/help/dictionary/vocabul.htm

**Comment by AnishShah** [comment]
Hi. If anybody else is not working on this, then can I work on this?

**Comment by malmaud** [comment]
@AnishShah You sure can - it would be appreciated.

**Comment by catchmrbharath** [comment]
@AnishShah Are you still working on this?

**Comment by rawls238** [comment]
can probably close this since this has been moved out of Base?

**Comment by tkelman** [comment]
Yep, good call, closed by #13897. Could be reopened as a feature request at Combinatorics.jl though.
